lib/core-server/src/build-static.ts

```diff
@@ -114,10 +115,10 @@ export async function buildStaticStandalone(options: CLIOptions & LoadOptions &
     logConfig('Manager webpack config', await managerBuilder.getConfig(fullOptions));
   }
 
-  const core = await presets.apply<{ builder?: string }>('core');
-
+  const core = await presets.apply<CoreConfig | undefined>('core');
+  const builderName = typeof core?.builder === 'string' ? core.builder : core?.builder?.name;
   const { getPrebuiltDir } =
-    core?.builder === 'webpack5'
+    builderName === 'webpack5'
```

Comment:
We probably want to assert that it's either `webpack5` or `webpack4` here, just to be safe?